addons/web_editor/static/src/xml/add_snippet_dialog.xml

```diff
@@ -0,0 +1,56 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<templates xml:space="preserve">
+
+<t t-name="web_editor.AddSnippetDialog">
+    <Dialog
+        title="title"
+        contentClass="'o_add_snippet_dialog h-100'"
+        footer="false"
+        size="'xl'">
+        <div class="overflow-hidden w-100">
+            <div class="d-flex w-100 h-100 vertical flex-row">
+                <ul class="nav nav-pills flex-column flex-nowrap overflow-y-auto border-end" t-ref="snippetGroupTabs">
+                <div class="input-group p-2 bg-100 border-bottom">
+                    <input type="search" class="form-control bg-white" placeholder="Search for a block"
+                        aria-label="Search for a block" t-model="state.search" t-ref="search-input"/>
+                    <span class="input-group-text border-0">
+                        <i class="fa fa-search"></i>
+                    </span>
+                </div>
+                    <t t-if="!state.search" t-foreach="snippetGroups" t-as="snippetGroup" t-key="snippetGroup.name">
+                        <li class="nav-item border-bottom">
+                            <a class="nav-link p-3 rounded-0"
+                                t-att-class="{ 'active': this.state.groupSelected === snippetGroup.name}"
+                                href="#"
+                                role="tab"
```

Comment:
A `role="tab"` without a `tablist` and `tabpanel` isn't correct (+ this kind of element needs some more attributes and the rest to be WCAG-compliant, see the [aria patterns example](https://www.w3.org/WAI/ARIA/apg/patterns/tabs)). Moreover, the `tab` should be an [accessibility child](https://w3c.github.io/aria/#dfn-accessibility-child) of the `tablist`, which means if the tablist is the `ul`, children of `li` aren't "accessibility children" (because `li` has another role by default, i.e. `listitem`).

So either the structure should be rewritten without `ul`, `li`, and just a `<div role="tablist">` with `<button role="tab">` as children, or the `role="tab"` should be on the `li` directly, not the inner button (but this might have other implications, I'm not sure).

Reply:
Note that we can consider it ok-ish for the merge, and work on that afterwards.

Reply:
TODO: after merge

Reply:
At the end, I made some changes (`list-group` div and  `list-group-item` button)